.ebextensions/nginx.config

```diff
@@ -0,0 +1,232 @@
+          application/x-java-archive-diff       jardiff;
+          application/x-java-jnlp-file          jnlp;
+          application/x-makeself                run;
+          application/x-opera-extension         oex;
+          application/x-perl                    pl pm;
+          application/x-pilot                   pdb prc;
+          application/x-rar-compressed          rar;
+          application/x-redhat-package-manager  rpm;
+          application/x-sea                     sea;
+          application/x-shockwave-flash         swf;
+          application/x-stuffit                 sit;
+          application/x-tcl                     tcl tk;
+          application/x-x509-ca-cert            crt der pem;
+          application/x-xpinstall               xpi;
+          application/xhtml+xml                 xhtml;
+          application/xslt+xml                  xsl;
+          application/zip                       zip;
+          text/css                              css;
+          text/csv                              csv;
+          text/html                             htm html shtml;
+          text/markdown                         md;
+          text/mathml                           mml;
+          text/plain                            txt;
+          text/vcard                            vcard vcf;
+          text/vnd.rim.location.xloc            xloc;
+          text/vnd.sun.j2me.app-descriptor      jad;
+          text/vnd.wap.wml                      wml;
+          text/vtt                              vtt;
+          text/x-component                      htc;
+
+      }
+
+  "/etc/nginx/extras/directive-only/extra-security.conf":
+    mode: "000644"
+    owner: root
+    group: root
+    content: |
+      # The X-Frame-Options header indicates whether a browser should be allowed
+      # to render a page within a frame or iframe.
+      add_header X-Frame-Options SAMEORIGIN always;
+
+      # MIME type sniffing security protection
+      #	There are very few edge cases where you wouldn't want this enabled.
+      add_header X-Content-Type-Options nosniff always;
+
+      # The X-XSS-Protection header is used by Internet Explorer version 8+
+      # The header instructs IE to enable its inbuilt anti-cross-site scripting filter.
+      add_header X-XSS-Protection "1; mode=block" always;
+
+      # with Content Security Policy (CSP) enabled (and a browser that supports it (http://caniuse.com/#feat=contentsecuritypolicy),
+      # you can tell the browser that it can only download content from the domains you explicitly allow
+      # CSP can be quite difficult to configure, and cause real issues if you get it wrong
+      # There is website that helps you generate a policy here http://cspisawesome.com/
+      add_header Content-Security-Policy "default-src 'self'; script-src 'self' 'unsafe-inline' https://dap.digitalgov.gov https://www.google-analytics.com; img-src 'self' *.tile.osm.org; frame-ancestors 'none';" always;
```

Comment:
For more on Content Security Policy headers, check out https://content-security-policy.com.

Reply:
This CSP should by default allow requests from the same origin (`'self'`) with customizations for JavaScript (`'self'` plus inline `<script>` elements and `<script>` elements coming from the DAP and Google Analytics) and for images (`'self'` plus OpenStreetMap-provided map tiles). It also prevents the site from being embedded elsewhere:

> `frame-ancestors`: Defines valid sources for embedding the resource using `<frame>` `<iframe>` `<object>` `<embed>` `<applet>`. Setting this directive to `'none'` should be roughly equivalent to `X-Frame-Options: DENY`.